src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -498,5 +500,66 @@ private static void ValidateRoleDefinition(PSRoleDefinition roleDefinition)
                 throw new ArgumentException(ProjectResources.InvalidActions);
             }
         }
+
+        public static void ValidateScope(string scope)
+        {
+
+            if (scope.Length == 0 || !scope.StartsWith("/"))
+            {
+                throw new ArgumentException(string.Format(ProjectResources.ScopeShouldBeginWithSubscriptionsOrProviders, scope));
+            }
+
+            // "/" is a valid scope.
+            if (string.Compare(scope, "/") == 0)
+            {
+                return;
+            }
+
+            var parts = scope.Substring(1).Split('/');   // Skip the leading '/'
+
+            if (parts.Contains(string.Empty))
+            {
+                throw new ArgumentException(string.Format(ProjectResources.ScopeShouldHaveNoEmptyPart, scope));
+            }
+
+            int count = parts.Count();
+
+            if (count % 2 != 0)
+            {
+                throw new ArgumentException(string.Format(ProjectResources.ScopeShouldHaveEvenNumberOfParts, scope));
+            }
+
+            // For Scope that on tenant level, without subscriptions, should start with "providers"
+            // in this case we only ensure that the number of parts is even.
+            if (string.Compare(parts[0], "providers", true) == 0)
+            {
+                return;
+            }
+            else if (string.Compare(parts[0], "subscriptions", true) != 0)
+            {
+                throw new ArgumentException(string.Format(ProjectResources.ScopeShouldBeginWithSubscriptionsOrProviders, scope));
+            }
+
+            // Now check the case that scope begins with '/subscriptions'
+
+            if (count >= 4 && string.Compare(parts[2], "resourceGroups", true) != 0)
```

Comment:
All these rules make sense to me. Is there a way to make sure that are there any current role assignments which are at scope which don't adhere to these rules.

At the end of the day you guys are the master of role assignments and the best judge for what the valid scope is and what is not. 

One way would be to add the same validation in the RP as well for the scope and make sure that both the validations in service and at client side matches.